smarts/core/utils/open_drive_exception.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2022. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+
+class OpenDRIVEException(Exception):
+    """An exception raised if opendrive utilities are required but not available."""
+
+    @classmethod
+    def required_to(cls, thing):
+        """Generate an instance of this exception that describes what can be done to remove the exception"""
+        return cls(
+            f"""OpenDRIVE Package is required to simulate {thing}.
```

Comment:
```suggestion
            f"""OpenDRIVE Package is required to {thing}.
```

Reply:
I did a bit more and moved both `RayException` and `OpenDriveException` to `smarts/core/utils/custom_exceptions.py`. This helps with cleaning up the example imports as well.